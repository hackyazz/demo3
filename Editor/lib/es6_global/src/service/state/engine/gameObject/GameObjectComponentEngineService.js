

import * as GameObjectAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/GameObjectAPI.js";
import * as GetComponentGameObjectService$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/service/record/main/gameObject/GetComponentGameObjectService.js";

function getLightMaterialComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getLightMaterialComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function getBasicMaterialComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getBasicMaterialComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function getMeshRendererComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getMeshRendererComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

var unsafeGetGeometryComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectGeometryComponent;

function getGeometryComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getGeometryComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function hasLightComponent(gameObject, engineState) {
  if (GameObjectAPI$Wonderjs.hasGameObjectDirectionLightComponent(gameObject, engineState)) {
    return true;
  } else {
    return GameObjectAPI$Wonderjs.hasGameObjectPointLightComponent(gameObject, engineState);
  }
}

var hasLightMaterialComponent = GameObjectAPI$Wonderjs.hasGameObjectLightMaterialComponent;

var unsafeGetLightMaterialComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectLightMaterialComponent;

var addLightMaterialComponent = GameObjectAPI$Wonderjs.addGameObjectLightMaterialComponent;

var disposeLightMaterialComponent = GameObjectAPI$Wonderjs.disposeGameObjectLightMaterialComponent;

var removeLightMaterialComponent = GameObjectAPI$Wonderjs.removeGameObjectLightMaterialComponent;

var hasBasicMaterialComponent = GameObjectAPI$Wonderjs.hasGameObjectBasicMaterialComponent;

var unsafeGetBasicMaterialComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectBasicMaterialComponent;

var addBasicMaterialComponent = GameObjectAPI$Wonderjs.addGameObjectBasicMaterialComponent;

var disposeBasicMaterialComponent = GameObjectAPI$Wonderjs.disposeGameObjectBasicMaterialComponent;

var removeBasicMaterialComponent = GameObjectAPI$Wonderjs.removeGameObjectBasicMaterialComponent;

var addMeshRendererComponent = GameObjectAPI$Wonderjs.addGameObjectMeshRendererComponent;

var unsafeGetMeshRendererComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectMeshRendererComponent;

var hasMeshRendererComponent = GameObjectAPI$Wonderjs.hasGameObjectMeshRendererComponent;

var disposeMeshRendererComponent = GameObjectAPI$Wonderjs.disposeGameObjectMeshRendererComponent;

var disposeGeometryComponent = GameObjectAPI$Wonderjs.disposeGameObjectGeometryComponent;

var removeGeometryComponent = GameObjectAPI$Wonderjs.removeGameObjectGeometryComponent;

var hasGeometryComponent = GameObjectAPI$Wonderjs.hasGameObjectGeometryComponent;

var addGeometryComponent = GameObjectAPI$Wonderjs.addGameObjectGeometryComponent;

var addPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.addGameObjectPerspectiveCameraProjectionComponent;

var hasPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.hasGameObjectPerspectiveCameraProjectionComponent;

var unsafeGetPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectPerspectiveCameraProjectionComponent;

var disposePerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.disposeGameObjectPerspectiveCameraProjectionComponent;

var addBasicCameraViewComponent = GameObjectAPI$Wonderjs.addGameObjectBasicCameraViewComponent;

var hasBasicCameraViewComponent = GameObjectAPI$Wonderjs.hasGameObjectBasicCameraViewComponent;

var unsafeGetBasicCameraViewComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectBasicCameraViewComponent;

var disposeBasicCameraViewComponent = GameObjectAPI$Wonderjs.disposeGameObjectBasicCameraViewComponent;

var unsafeGetTransformComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectTransformComponent;

var hasTransformComponent = GameObjectAPI$Wonderjs.hasGameObjectTransformComponent;

var unsafeGetSourceInstanceComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectSourceInstanceComponent;

var hasSourceInstanceComponent = GameObjectAPI$Wonderjs.hasGameObjectSourceInstanceComponent;

var addSourceInstanceComponent = GameObjectAPI$Wonderjs.addGameObjectSourceInstanceComponent;

var addDirectionLightComponent = GameObjectAPI$Wonderjs.addGameObjectDirectionLightComponent;

var disposeDirectionLightComponent = GameObjectAPI$Wonderjs.disposeGameObjectDirectionLightComponent;

var unsafeGetDirectionLightComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectDirectionLightComponent;

var hasDirectionLightComponent = GameObjectAPI$Wonderjs.hasGameObjectDirectionLightComponent;

var addPointLightComponent = GameObjectAPI$Wonderjs.addGameObjectPointLightComponent;

var disposePointLightComponent = GameObjectAPI$Wonderjs.disposeGameObjectPointLightComponent;

var unsafeGetPointLightComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectPointLightComponent;

var hasPointLightComponent = GameObjectAPI$Wonderjs.hasGameObjectPointLightComponent;

var addArcballCameraControllerComponent = GameObjectAPI$Wonderjs.addGameObjectArcballCameraControllerComponent;

var disposeArcballCameraControllerComponent = GameObjectAPI$Wonderjs.disposeGameObjectArcballCameraControllerComponent;

var unsafeGetArcballCameraControllerComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectArcballCameraControllerComponent;

var hasArcballCameraControllerComponent = GameObjectAPI$Wonderjs.hasGameObjectArcballCameraControllerComponent;

var getAllLightMaterialComponents = GameObjectAPI$Wonderjs.getAllLightMaterialComponents;

var getAllDirectionLightComponents = GameObjectAPI$Wonderjs.getAllDirectionLightComponents;

var getAllBasicCameraViewComponents = GameObjectAPI$Wonderjs.getAllBasicCameraViewComponents;

var getAllArcballCameraControllerComponents = GameObjectAPI$Wonderjs.getAllArcballCameraControllerComponents;

var getAllPerspectiveCameraProjectionComponents = GameObjectAPI$Wonderjs.getAllPerspectiveCameraProjectionComponents;

var getAllGeometryComponents = GameObjectAPI$Wonderjs.getAllGeometryComponents;

export {
  hasLightMaterialComponent ,
  getLightMaterialComponent ,
  unsafeGetLightMaterialComponent ,
  addLightMaterialComponent ,
  disposeLightMaterialComponent ,
  removeLightMaterialComponent ,
  hasBasicMaterialComponent ,
  getBasicMaterialComponent ,
  unsafeGetBasicMaterialComponent ,
  addBasicMaterialComponent ,
  disposeBasicMaterialComponent ,
  removeBasicMaterialComponent ,
  addMeshRendererComponent ,
  getMeshRendererComponent ,
  unsafeGetMeshRendererComponent ,
  hasMeshRendererComponent ,
  disposeMeshRendererComponent ,
  unsafeGetGeometryComponent ,
  getGeometryComponent ,
  disposeGeometryComponent ,
  removeGeometryComponent ,
  hasGeometryComponent ,
  addGeometryComponent ,
  addPerspectiveCameraProjectionComponent ,
  hasPerspectiveCameraProjectionComponent ,
  unsafeGetPerspectiveCameraProjectionComponent ,
  disposePerspectiveCameraProjectionComponent ,
  addBasicCameraViewComponent ,
  hasBasicCameraViewComponent ,
  unsafeGetBasicCameraViewComponent ,
  disposeBasicCameraViewComponent ,
  unsafeGetTransformComponent ,
  hasTransformComponent ,
  unsafeGetSourceInstanceComponent ,
  hasSourceInstanceComponent ,
  addSourceInstanceComponent ,
  addDirectionLightComponent ,
  disposeDirectionLightComponent ,
  unsafeGetDirectionLightComponent ,
  hasDirectionLightComponent ,
  addPointLightComponent ,
  disposePointLightComponent ,
  unsafeGetPointLightComponent ,
  hasPointLightComponent ,
  hasLightComponent ,
  addArcballCameraControllerComponent ,
  disposeArcballCameraControllerComponent ,
  unsafeGetArcballCameraControllerComponent ,
  hasArcballCameraControllerComponent ,
  getAllLightMaterialComponents ,
  getAllDirectionLightComponents ,
  getAllBasicCameraViewComponents ,
  getAllArcballCameraControllerComponents ,
  getAllPerspectiveCameraProjectionComponents ,
  getAllGeometryComponents ,
  
}
/* GameObjectAPI-Wonderjs Not a pure module */
