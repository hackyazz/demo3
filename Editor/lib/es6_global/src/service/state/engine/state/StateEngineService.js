

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
