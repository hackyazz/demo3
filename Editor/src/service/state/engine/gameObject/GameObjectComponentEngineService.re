open Wonderjs;

open StateDataMainType;

/* light material */

let hasLightMaterialComponent = GameObjectAPI.hasGameObjectLightMaterialComponent;

let getLightMaterialComponent = (gameObject, engineState) =>
  GetComponentGameObjectService.getLightMaterialComponent(.
    gameObject,
    engineState.gameObjectRecord,
  );

let unsafeGetLightMaterialComponent = GameObjectAPI.unsafeGetGameObjectLightMaterialComponent;

let addLightMaterialComponent = GameObjectAPI.addGameObjectLightMaterialComponent;

let disposeLightMaterialComponent = GameObjectAPI.disposeGameObjectLightMaterialComponent;

let removeLightMaterialComponent = GameObjectAPI.removeGameObjectLightMaterialComponent;

/* basic material */

let hasBasicMaterialComponent = GameObjectAPI.hasGameObjectBasicMaterialComponent;

let getBasicMaterialComponent = (gameObject, engineState) =>
  GetComponentGameObjectService.getBasicMaterialComponent(.
    gameObject,
    engineState.gameObjectRecord,
  );

let unsafeGetBasicMaterialComponent = GameObjectAPI.unsafeGetGameObjectBasicMaterialComponent;

let addBasicMaterialComponent = GameObjectAPI.addGameObjectBasicMaterialComponent;

let disposeBasicMaterialComponent = GameObjectAPI.disposeGameObjectBasicMaterialComponent;

let removeBasicMaterialComponent = GameObjectAPI.removeGameObjectBasicMaterialComponent;

/* mesh renderer */

let addMeshRendererComponent = GameObjectAPI.addGameObjectMeshRendererComponent;

let getMeshRendererComponent = (gameObject, engineState) =>
  GetComponentGameObjectService.getMeshRendererComponent(.
    gameObject,
    engineState.gameObjectRecord,
  );

let unsafeGetMeshRendererComponent = GameObjectAPI.unsafeGetGameObjectMeshRendererComponent;

let hasMeshRendererComponent = GameObjectAPI.hasGameObjectMeshRendererComponent;

let disposeMeshRendererComponent = GameObjectAPI.disposeGameObjectMeshRendererComponent;

/* geometry */

let unsafeGetGeometryComponent = (gameObject, engineState) =>
  engineState
  |> GameObjectAPI.unsafeGetGameObjectGeometryComponent(gameObject);

let getGeometryComponent = (gameObject, engineState) =>
  GetComponentGameObjectService.getGeometryComponent(.
    gameObject,
    engineState.gameObjectRecord,
  );

let disposeGeometryComponent = GameObjectAPI.disposeGameObjectGeometryComponent;

let removeGeometryComponent = GameObjectAPI.removeGameObjectGeometryComponent;

let hasGeometryComponent = GameObjectAPI.hasGameObjectGeometryComponent;

let addGeometryComponent = GameObjectAPI.addGameObjectGeometryComponent;
/* perspective camera projection */

let addPerspectiveCameraProjectionComponent = GameObjectAPI.addGameObjectPerspectiveCameraProjectionComponent;

let hasPerspectiveCameraProjectionComponent = GameObjectAPI.hasGameObjectPerspectiveCameraProjectionComponent;

let unsafeGetPerspectiveCameraProjectionComponent = GameObjectAPI.unsafeGetGameObjectPerspectiveCameraProjectionComponent;

let disposePerspectiveCameraProjectionComponent = GameObjectAPI.disposeGameObjectPerspectiveCameraProjectionComponent;
/* camera view */

let addBasicCameraViewComponent = GameObjectAPI.addGameObjectBasicCameraViewComponent;

let hasBasicCameraViewComponent = GameObjectAPI.hasGameObjectBasicCameraViewComponent;

let unsafeGetBasicCameraViewComponent = GameObjectAPI.unsafeGetGameObjectBasicCameraViewComponent;

let disposeBasicCameraViewComponent = GameObjectAPI.disposeGameObjectBasicCameraViewComponent;
/* transform */

let unsafeGetTransformComponent = GameObjectAPI.unsafeGetGameObjectTransformComponent;

let hasTransformComponent = GameObjectAPI.hasGameObjectTransformComponent;

/* source instance */
let unsafeGetSourceInstanceComponent = GameObjectAPI.unsafeGetGameObjectSourceInstanceComponent;

let hasSourceInstanceComponent = GameObjectAPI.hasGameObjectSourceInstanceComponent;

let addSourceInstanceComponent = GameObjectAPI.addGameObjectSourceInstanceComponent;

/* direction light */

let addDirectionLightComponent = GameObjectAPI.addGameObjectDirectionLightComponent;

let disposeDirectionLightComponent = GameObjectAPI.disposeGameObjectDirectionLightComponent;

let unsafeGetDirectionLightComponent = GameObjectAPI.unsafeGetGameObjectDirectionLightComponent;

let hasDirectionLightComponent = GameObjectAPI.hasGameObjectDirectionLightComponent;

/* point light */
let addPointLightComponent = GameObjectAPI.addGameObjectPointLightComponent;

let disposePointLightComponent = GameObjectAPI.disposeGameObjectPointLightComponent;

let unsafeGetPointLightComponent = GameObjectAPI.unsafeGetGameObjectPointLightComponent;

let hasPointLightComponent = GameObjectAPI.hasGameObjectPointLightComponent;

let hasLightComponent = (gameObject, engineState) =>
  hasDirectionLightComponent(gameObject, engineState)
  || hasPointLightComponent(gameObject, engineState);

/* arcball camera */

let addArcballCameraControllerComponent = GameObjectAPI.addGameObjectArcballCameraControllerComponent;

let disposeArcballCameraControllerComponent = GameObjectAPI.disposeGameObjectArcballCameraControllerComponent;

let unsafeGetArcballCameraControllerComponent = GameObjectAPI.unsafeGetGameObjectArcballCameraControllerComponent;

let hasArcballCameraControllerComponent = GameObjectAPI.hasGameObjectArcballCameraControllerComponent;

/* all components */

let getAllLightMaterialComponents = GameObjectAPI.getAllLightMaterialComponents;

let getAllDirectionLightComponents = GameObjectAPI.getAllDirectionLightComponents;

let getAllBasicCameraViewComponents = GameObjectAPI.getAllBasicCameraViewComponents;

let getAllArcballCameraControllerComponents = GameObjectAPI.getAllArcballCameraControllerComponents;

let getAllPerspectiveCameraProjectionComponents = GameObjectAPI.getAllPerspectiveCameraProjectionComponents;

let getAllGeometryComponents = GameObjectAPI.getAllGeometryComponents;