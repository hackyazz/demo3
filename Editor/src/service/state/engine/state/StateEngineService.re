let deepCopyForRestore = Wonderjs.StateAPI.deepCopyForRestore;

let createState = Wonderjs.StateAPI.createState;

let createStateData = Wonderjs.StateAPI.createStateData;

let getStateFromData = Wonderjs.StateAPI.getStateFromData;

let unsafeGetState = Wonderjs.StateAPI.unsafeGetState;

let setStateToData = Wonderjs.StateAPI.setStateToData;

let setState = Wonderjs.StateAPI.setState;

let setIsDebug = Wonderjs.StateAPI.setIsDebug;