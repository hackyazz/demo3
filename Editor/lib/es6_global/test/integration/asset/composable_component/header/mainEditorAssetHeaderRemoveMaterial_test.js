

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as ReactTestTool$WonderEditor from "../../../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../../../tool/GameObjectTool.js";
import * as BuildComponentTool$WonderEditor from "../../../../tool/BuildComponentTool.js";
import * as StateEngineService$WonderEditor from "../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorAssetTool$WonderEditor from "../../tool/MainEditorAssetTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../tool/MainEditorSceneTool.js";
import * as MainEditorAssetIdTool$WonderEditor from "../../tool/MainEditorAssetIdTool.js";
import * as MainEditorMaterialTool$WonderEditor from "../../../inspector/composable_component/sceneTree_inspector/renderGroup/material/tool/MainEditorMaterialTool.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../tool/MainEditorAssetTreeTool.js";
import * as MainEditorAssetMaterialNodeTool$WonderEditor from "../../tool/MainEditorAssetMaterialNodeTool.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../src/service/state/engine/GameObjectComponentEngineService.js";
import * as MainEditorAssetHeaderOperateNodeTool$WonderEditor from "../../tool/MainEditorAssetHeaderOperateNodeTool.js";

describe("MainEditorAssetHeader->remove material", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorAssetTool$WonderEditor.initAssetTree);
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("\n        select material;\n        click remove-button;\n            ", (function () {
                return Wonder_jest.test("should remove it from assetTreeRoot", (function () {
                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* buildOneMaterialAssetTree */0], /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeMaterialNode(undefined, undefined, Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* getFirstMaterialNodeId */2], assetTreeData), /* () */0);
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetTree(/* () */0));
                            }));
              }));
        describe("\n        add material m1;\n        select sceneTree->gameObject g1;\n        g1 change material to m1;\n        remove m1;\n        ", (function () {
                Wonder_jest.test("select material group shouldn't contain removed material assets", (function () {
                        MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                        var addedMaterialNodeId = MainEditorAssetIdTool$WonderEditor.getNewAssetId(undefined, /* () */0);
                        var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                        var gameObject1 = MainEditorSceneTool$WonderEditor.getFirstCube(engineState);
                        var sourceMaterial1 = GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(gameObject1, engineState);
                        MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                        var materialComponent = MainEditorAssetMaterialNodeTool$WonderEditor.getMaterialComponent(addedMaterialNodeId, undefined, /* () */0);
                        MainEditorMaterialTool$WonderEditor.changeMaterial(sourceMaterial1, /* LightMaterial */1, materialComponent, /* LightMaterial */1, addedMaterialNodeId, gameObject1, undefined, undefined, /* () */0);
                        MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeMaterialNode(undefined, undefined, addedMaterialNodeId, /* () */0);
                        GameObjectTool$WonderEditor.setCurrentSceneTreeNode(gameObject1);
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMaterial(gameObject1, undefined, undefined, true, /* () */0));
                      }));
                describe("g1 should use default light material", (function () {
                        Wonder_jest.test("test snapshot", (function () {
                                MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                var addedMaterialNodeId = MainEditorAssetIdTool$WonderEditor.getNewAssetId(undefined, /* () */0);
                                var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                var gameObject1 = MainEditorSceneTool$WonderEditor.getFirstCube(engineState);
                                var sourceMaterial1 = GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(gameObject1, engineState);
                                MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                                var materialComponent = MainEditorAssetMaterialNodeTool$WonderEditor.getMaterialComponent(addedMaterialNodeId, undefined, /* () */0);
                                MainEditorMaterialTool$WonderEditor.changeMaterial(sourceMaterial1, /* LightMaterial */1, materialComponent, /* LightMaterial */1, addedMaterialNodeId, gameObject1, undefined, undefined, /* () */0);
                                MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeMaterialNode(undefined, undefined, addedMaterialNodeId, /* () */0);
                                GameObjectTool$WonderEditor.setCurrentSceneTreeNode(gameObject1);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMaterial(gameObject1, undefined, undefined, true, /* () */0));
                              }));
                        return Wonder_jest.test("g1's material component shouldn't be m1", (function () {
                                      MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                      var addedMaterialNodeId = MainEditorAssetIdTool$WonderEditor.getNewAssetId(undefined, /* () */0);
                                      var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                      var gameObject1 = MainEditorSceneTool$WonderEditor.getFirstCube(engineState);
                                      var sourceMaterial1 = GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(gameObject1, engineState);
                                      MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                                      var materialComponent = MainEditorAssetMaterialNodeTool$WonderEditor.getMaterialComponent(addedMaterialNodeId, undefined, /* () */0);
                                      MainEditorMaterialTool$WonderEditor.changeMaterial(sourceMaterial1, /* LightMaterial */1, materialComponent, /* LightMaterial */1, addedMaterialNodeId, gameObject1, undefined, undefined, /* () */0);
                                      MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeMaterialNode(undefined, undefined, addedMaterialNodeId, /* () */0);
                                      var engineState$1 = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                      return Wonder_jest.Expect[/* toEqual */12](materialComponent)(Wonder_jest.Expect[/* not_ */22](Wonder_jest.Expect[/* expect */0](GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(gameObject1, engineState$1))));
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
