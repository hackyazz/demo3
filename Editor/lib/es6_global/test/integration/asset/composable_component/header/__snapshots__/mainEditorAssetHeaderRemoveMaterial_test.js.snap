// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorAssetHeader->remove material 
        add material m1;
        select sceneTree->gameObject g1;
        g1 change material to m1;
        remove m1;
         g1 should use default light material test snapshot 1`] = `
<article
  className="wonder-inspector-material"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Material
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Light-Material
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="select-item-header"
      >
        Material
      </div>
      <div
        className="select-item-body"
      >
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Wonder-Default-Basic-Material
        </div>
        <div
          className="select-item-content select-item-active"
          onClick={[Function]}
        >
          Wonder-Default-Light-Material
        </div>
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
  <div
    className="material-value"
  >
    <article
      className="inspector-item"
    >
      <div
        className="item-header"
      >
        <span
          className=""
        >
          Type
        </span>
      </div>
      <div
        className="item-content"
      >
        <select
          onChange={[Function]}
          value="1"
        >
          <option
            value="0"
          >
            basic_material
          </option>
          <option
            value="1"
          >
            light_material
          </option>
        </select>
      </div>
    </article>
    <div
      className=""
    >
      <article
        className="wonder-light-material"
      >
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffcuse color
          </div>
          <div
            className="item-content item-color"
          >
            <div
              className="color-hex"
              onClick={[Function]}
              style={
                Object {
                  "background": "#ffffff",
                }
              }
            />
            <div
              className="color-select"
              onClick={[Function]}
            >
              <img
                src="./public/img/color.png"
              />
            </div>
          </div>
        </article>
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffuse map
          </div>
          <div
            className="item-content item-texture"
          >
            <div
              className="texture-img"
              onClick={[Function]}
              style={
                Object {
                  "opacity": "1",
                }
              }
            >
              <div
                className="img-dragBg"
                onDragEnter={[Function]}
                onDragLeave={[Function]}
                onDragOver={[Function]}
                onDrop={[Function]}
              />
            </div>
            <button
              className="texture-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </article>
        <article
          className="inspector-item wonder-float-input"
        >
          <div
            className="item-header component-label"
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            Shininess
          </div>
          <div
            className="item-content"
          >
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="32"
            />
          </div>
        </article>
      </article>
    </div>
    <div
      className="material-shade"
    />
  </div>
</article>
`;

exports[`MainEditorAssetHeader->remove material 
        add material m1;
        select sceneTree->gameObject g1;
        g1 change material to m1;
        remove m1;
         select material group shouldn't contain removed material assets 1`] = `
<article
  className="wonder-inspector-material"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Material
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Light-Material
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="select-item-header"
      >
        Material
      </div>
      <div
        className="select-item-body"
      >
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Wonder-Default-Basic-Material
        </div>
        <div
          className="select-item-content select-item-active"
          onClick={[Function]}
        >
          Wonder-Default-Light-Material
        </div>
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
  <div
    className="material-value"
  >
    <article
      className="inspector-item"
    >
      <div
        className="item-header"
      >
        <span
          className=""
        >
          Type
        </span>
      </div>
      <div
        className="item-content"
      >
        <select
          onChange={[Function]}
          value="1"
        >
          <option
            value="0"
          >
            basic_material
          </option>
          <option
            value="1"
          >
            light_material
          </option>
        </select>
      </div>
    </article>
    <div
      className=""
    >
      <article
        className="wonder-light-material"
      >
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffcuse color
          </div>
          <div
            className="item-content item-color"
          >
            <div
              className="color-hex"
              onClick={[Function]}
              style={
                Object {
                  "background": "#ffffff",
                }
              }
            />
            <div
              className="color-select"
              onClick={[Function]}
            >
              <img
                src="./public/img/color.png"
              />
            </div>
          </div>
        </article>
        <article
          className="inspector-item"
        >
          <div
            className="item-header"
          >
            Diffuse map
          </div>
          <div
            className="item-content item-texture"
          >
            <div
              className="texture-img"
              onClick={[Function]}
              style={
                Object {
                  "opacity": "1",
                }
              }
            >
              <div
                className="img-dragBg"
                onDragEnter={[Function]}
                onDragLeave={[Function]}
                onDragOver={[Function]}
                onDrop={[Function]}
              />
            </div>
            <button
              className="texture-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </article>
        <article
          className="inspector-item wonder-float-input"
        >
          <div
            className="item-header component-label"
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
          >
            Shininess
          </div>
          <div
            className="item-content"
          >
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="32"
            />
          </div>
        </article>
      </article>
    </div>
    <div
      className="material-shade"
    />
  </div>
</article>
`;

exports[`MainEditorAssetHeader->remove material 
        select material;
        click remove-button;
             should remove it from assetTreeRoot 1`] = `
<article
  className="wonder-asset-assetTree"
>
  <ul
    className="wonder-tree-node"
  >
    <li>
      <div
        className="item-triangle"
      />
      <img
        className="treeNode-icon"
        src="./public/img/package.png"
      />
      <div
        className="draggable-container select-not-active"
        draggable={true}
        onClick={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        style={Object {}}
      >
        Assets
      </div>
    </li>
  </ul>
</article>
`;
